eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Utxo.hs

```diff
@@ -392,11 +504,26 @@ alonzoToConwayUtxoPredFailure = \case
       OutputTooBigUTxO $ map toRestricted xs
   Alonzo.InsufficientCollateral c1 c2 -> InsufficientCollateral c1 c2
   Alonzo.ScriptsNotPaidUTxO u -> ScriptsNotPaidUTxO u
-  Alonzo.ExUnitsTooBigUTxO e1 e2 -> ExUnitsTooBigUTxO e1 e2
+  Alonzo.ExUnitsTooBigUTxO e1 e2 ->
+    ExUnitsTooBigUTxO
+      Mismatch
+        { mismatchSupplied = e2
+        , mismatchExpected = e1
+        }
   Alonzo.CollateralContainsNonADA v -> CollateralContainsNonADA v
-  Alonzo.WrongNetworkInTxBody nid nidb -> WrongNetworkInTxBody nid nidb
+  Alonzo.WrongNetworkInTxBody nid nidInTx ->
+    WrongNetworkInTxBody
+      Mismatch
+        { mismatchSupplied = nidInTx
+        , mismatchExpected = nid
+        }
   Alonzo.OutsideForecast sno -> OutsideForecast sno
-  Alonzo.TooManyCollateralInputs n1 n2 -> TooManyCollateralInputs n1 n2
+  Alonzo.TooManyCollateralInputs maxI suppliedI ->
+    TooManyCollateralInputs
+      Mismatch
```

Comment:
Maybe have a comment about reversal here?

Reply:
I intended the comment with respect to the `Mismatch` type, as in, when the serialisation needs to swap the order. Here, we are just translation from the older to the newer era, and the `Alonzo` type will soon be made to use `Mismatch` in the following PRs. So I omitted the comment unless it was due to serialisation.